src/bin/stratisd.rs

```diff
@@ -511,19 +499,24 @@ fn main() {
 
     // Using a let-expression here so that the scope of the lock file
     // is the rest of the block.
-    let lock_file = trylock_pid_file();
-
-    let result = {
-        match lock_file {
-            Err(err) => Err(err),
-            Ok(_) => {
-                let log_handle = initialize_log(matches.is_present("debug"));
-                run(&matches, &log_handle)
-            }
+    let socket = match try_abstract_socket() {
```

Comment:
You'll want to drop the comment above, if it does not apply. It was necessary to extend the scope of the lock file. otherwise it got closed, and it was possible to start a new stratisd process. This was a problem for us, hence the comment.

I think https://docs.rs/nix/0.15.0/nix/unistd/fn.close.html is relevant here, as it is possible to drop the socket twice.

Reply:
But you have a raw fd, so I think that doesn't apply, actually, and you're only dropping it once.

Reply:
Yes, that message is a bit confusing. `RawFd` does not implement `Drop` so this is only applicable when calling `.as_raw_fd()` in the case of an object such as `File` which does implement `Drop`. I will remove the comment. That was an oversight on my part.